core/domain/user_services.py

```diff
@@ -1062,6 +1250,81 @@ def set_username(user_id, new_username):
     _save_user_settings(user_settings)
 
 
+def update_user_settings_and_auth_data(user_id, data):
+    """Updates PIN, display alias, subject interests and language preferences
+    for the user with the given user_id.
+
+    Args:
+        user_id: str. The unique ID of the user whose data has to be updated.
+        data: dict. The dictionary containing the new user data, whose keys
```

Comment:
Don't do this -- we try to never pass dicts around. If you need all this info, pass the args in separately. Or, declare methods on that domain object along the lines of update_pin() etc. and call those. See the pattern in exp_services.py.

Reply:
I have created a domain change object for UserSettings and UserAuthDetails - named as `UserDetailsChange`.